File: Source/Fantasy.Base/Tracking/TrackProperty.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Runtime.Serialization;

namespace Fantasy.Tracking
{
    // Use a data contract as illustrated in the sample below to add composite types to service operations
    [DataContract]
    public class TrackProperty
    {
        [DataMember]
        public string Name { get; set; }

        [DataMember]
        public string Value { get; set; }

        [DataMember]
        public string TypeName { get; set; }


        internal static TrackProperty Create(string name, object value)
        {
            TrackProperty rs = new TrackProperty() {Name = name};
            if(value != null)
            {
                rs.Value = value.ToString();
                rs.TypeName = value.GetType().FullName;
            }
            return rs;

        }

        internal static object ToObject(TrackProperty property)
        {
            if (property.TypeName != null)
            {
                Type t = Type.GetType(property.TypeName);
                return Convert.ChangeType(property.Value, t);
            }
            else
            {
                return null;
            }
        }
    }
}

File: branches/Stheno/AutoConverted/Fantasy.Base/Tracking/TrackProperty.java
package Fantasy.Tracking;

// Use a data contract as illustrated in the sample below to add composite types to service operations
//C# TO JAVA CONVERTER TODO TASK: Java annotations will not correspond to .NET attributes:
//[DataContract]
public class TrackProperty
{
//C# TO JAVA CONVERTER TODO TASK: Java annotations will not correspond to .NET attributes:
	//[DataMember]
	private String privateName;
	public final String getName()
	{
		return privateName;
	}
	public final void setName(String value)
	{
		privateName = value;
	}

//C# TO JAVA CONVERTER TODO TASK: Java annotations will not correspond to .NET attributes:
	//[DataMember]
	private String privateValue;
	public final String getValue()
	{
		return privateValue;
	}
	public final void setValue(String value)
	{
		privateValue = value;
	}

//C# TO JAVA CONVERTER TODO TASK: Java annotations will not correspond to .NET attributes:
	//[DataMember]
	private String privateTypeName;
	public final String getTypeName()
	{
		return privateTypeName;
	}
	public final void setTypeName(String value)
	{
		privateTypeName = value;
	}


	public static TrackProperty Create(String name, Object value)
	{
		TrackProperty tempVar = new TrackProperty();
		tempVar.setName(name);
		TrackProperty rs = tempVar;
		if(value != null)
		{
			rs.setValue(value.toString());
			rs.setTypeName(value.getClass().FullName);
		}
		return rs;

	}

	public static Object ToObject(TrackProperty property)
	{
		if (property.getTypeName() != null)
		{
			java.lang.Class t = java.lang.Class.forName(property.getTypeName());
			return Convert.ChangeType(property.getValue(), t);
		}
		else
		{
			return null;
		}
	}
}